homeassistant/components/godice/sensor.py

```diff
@@ -0,0 +1,132 @@
+"""Sensors for GoDice device."""
+
+import logging
+
+import godice
+
+from homeassistant import const
+from homeassistant.components.sensor import (
+    RestoreSensor,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DATA_DEVICE, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+ROLLED_NUMBER_SENSOR_DESCR = SensorEntityDescription(
+    key="rolled_number", state_class=SensorStateClass.MEASUREMENT
+)
+COLOR_SENSOR_DESCR = SensorEntityDescription(
+    key="color",
+    device_class=SensorDeviceClass.ENUM,
+    options=[color.name for color in godice.Color],
+)
+BATTERY_SENSOR_DESCR = SensorEntityDescription(
+    key="battery_level",
+    device_class=SensorDeviceClass.BATTERY,
+    native_unit_of_measurement=const.PERCENTAGE,
+    state_class=SensorStateClass.MEASUREMENT,
+)
```

Comment:
What you could also do here is this:
```suggestion
ENTITIES: dict[type[BaseSensor], SensorEntityDescription] = {
    DiceColorSensor: SensorEntityDescription(...),
    BatteryLevelSensor: SensorEntityDescription(...),
    DiceNumberSensor: SensorEntityDescription(...),
}

# In init:
async_add_entities(
    [
        entity_class(entry, description, dice)
        for entity_class, description in entity_descriptions.items()
    ]
)
```

Then you can get rid of the sensor_create_data in async_setup_entry as your dictionary will already hold the class and description pairs
```python
    sensor_create_data = [
        (DiceColorSensor, COLOR_SENSOR_DESCR),
        (BatteryLevelSensor, BATTERY_SENSOR_DESCR),
        (DiceNumberSensor, ROLLED_NUMBER_SENSOR_DESCR),
    ]
```

Reply:
I'd be good if I could put ENTITIES at the top of the file, but since Sensor classes are defined at the bottom they are not visible for reference at the top. So I can only put ENTITIES after Sensor definitions which as for me is not so readable

Reply:
You're right, in my case I put the constant at the bottom to fix this but you can also leave it like this if you don't want to do that